### DISCUSSION TITLE
5x5 manual mesh is lost when power restarted

### DISCUSSION THREAD

**Discussion by orangoo** [Jyers/Marlin#608] [Q&A] [answered]
Hi

I have successfully created and saved a 5x5 mesh.
I test this by printing 9 squares at 0.2mm layer height and during this initial print I use the Z offset from the tune menu available to make any minor adjustments. The print of the 9 squares are perfect.

I then turn off the printer, wait a few minutes and turn back on again.
I repeat the printing of the 9 squares and the nozzle is now too high above the bed.

I revisit the manual mesh bed levelling option and I can see at each of the 25 points the 'Z position' displays 0.20. Fortunately, I am able to use the 'Goto Mesh Value' option at the bottom of that menu, apply that figure to that particular point and move to the next point and repeat. At the end of the 25th point I save the mesh, print the 9 squares and it prints perfectly (in the same way as my initial print).

Is there something I'm doing wrong as to why the manual mesh is not using the saved mesh?

I have firmware 1.3.1 installed on the 4.2.2 board.

thanks

**Comment by Jyers** [comment, marked as answer]
If you go back into the leveling menu, it assumes you want to make a new mesh. The fact that it can go to the previous point shows that your mesh is properly saved, you do not need to go through the menu every time you power cycle the printer.

I am currently working on a new implementation of the leveling menu that will make this more clear.

**Reply by orangoo** [reply]
Thanks for the reply. 

Just to be clear here are the steps I am taking.

1) Perform manual mesh and save mesh
2) Head to Control > Store Settings and select. (I forgot to mention this earlier. Is this step necessary?)
3) Print an object - in my case the 9 squares to test for correct bed levelling at the 9 points and adjust using the Z offset from the Tune menu. The print is perfect.
4) Turn off printer (just as you would turn it off for the day)
5) Turn on printer
6) Select object to print - in this step I repeat the 9 squares test, but it can be any object to print.
Observe the 1st layer and I can see that the nozzle is definately higher than the saved mesh. As it continues its print I can definately see the first layer is not adhering to the board - I can run my finger over the initial purge line and it comes off.

I agree that the mesh is being saved as I can see it when I go back into the manual mesh option - so that's good. (Suggests to me that the saving to EEPROM is working).

Also - I have this in my start G-code:
G28 ; home all
M420 S1 ;Load Mesh Bed Level
(it's in the order you see above and I am using Prusa Slicer 2.3.0)

Very sorry if there is something obvious I am missing. Thanks very much for your help.

**Reply by Jyers** [reply]
There was a discussion regarding this kind of behavior that went on for a long while. I don't think we ever came to a conclusive answer though. Perhaps the new leveling menus will help to illuminate whatever is happening here. For now have you tried using the z offset to lower the nozzle while printing?

**Reply by orangoo** [reply]
Thanks for the suggestion. I haven't tried lowering the nozzle using the Z offset whilst printing. I'll give that go.
I do recall that there was something like this happening before in another thread.

I would've attempted to make changes to configuration.h to use RESTORE_LEVELING_AFTER_G28, but figured the M420 S1 command did the same thing.

It's odd that this issue seems to be only affecting me so far. Could I have a dud machine? thanks

**Reply by Jyers** [reply]
Check this out #212 You're definitely not alone.

**Reply by orangoo** [reply]
Thanks. This is the thread I read about a while back. I'll keep investigating and will let you know if I find anything of use. Very much appreciated for your assistance and feedback

**Reply by orangoo** [reply]
If I can make a suggestion (only if it is not to time consuming or difficult).

Regarding both the 3x3 and 5x5 manual mesh levelling points, would you be able to advise the exact co-ordinates (unless they are already displayed on the screen. If that's the case you can ignore this request and I can go and have a look)

Then I could manually insert the co-ordinates with the associated Z offset for each point into the configuation itself. Sorry, this is all quite new to me and I could be making life more difficult that it needs to be or going down the wrong path.

I read about it here:
https://marlinfw.org/docs/gcode/G029-mbl.html

thanks

**Reply by Jyers** [reply]
The coordinates are displayed on the bottom of the screen as usual. Also that is not the right version, manual mesh uses bilinear with the PROBE_MANUALLY option enabled. See https://marlinfw.org/docs/gcode/G029-abl-bilinear.html

**Reply by orangoo** [reply]
Thanks very much again. I'll give this a go when I have time as I'll be interested if this could be a workaround. I'll keep you posted.

**Reply by orangoo** [reply]
Here's the story so far.

I decided to see if using the 3x3 .bin file (for v1.3.1) would work and no - same end result where the manual mesh wasn't saving.
I made a note of the 9 probe points and the z values for my machine so that I could write them into the gcode. Had a lot on recently so only got a third of the way through. Anyway - I've just seen you have released 1.3.2. Decided to give that a whirl (also note I had the same issue someone else mentioned about loading the .bin file and the machine not updating, and having to change the filename in order for this to work).

Anyway - 1.3.2 and print has begun.

So far 25% into my print and it appears to be working. The saved manual mesh seems to be in use as I've made no changes.

Not sure what else I can say is maybe there was something in the .bin file that didn't want to use the saved manual mesh.

Thought you'd like to know. I'll continue to keep any eye on it but I think based on what I can see - your latest version seems to have resolved the issue.

thanks very much again

**Reply by Jyers** [reply]
Very interesting, I can't think what changed about how the mesh is saved but I'm glad it did!
